### ISSUE TITLE
Support the new .NET 10 LeftJoin operators

### ISSUE THREAD

**Issue by JohnGalt1717** [dotnet/efcore#12793]
Describe what is not working as expected.

Doing outer joins is with linq are horrible and make for a mess of code that makes it difficult if not impossible to debug.

I.e:

```csharp
from c in Context.Courses
join ci in Context.CourseItems on c.Id equals ci.CourseId

join tijoin in Context.TopicItems on ci.ItemId equals tijoin.TopicId into titrans
from ti in titrans.DefaultIfEmpty()
```

By adding leftjoin we'd get this:

```csharp
from c in Context.Courses
join ci in Context.CourseItems on c.Id equals ci.CourseId
leftjoin ti in Context.TopicItems on ci.ItemId equals ti.TopicId

```

Yes it really is that ugly every single time simply because there isn't a leftjoin operator that does the same.

Would this break current providers? No, because it can be converted first to the ugly syntax they already support if they don't support the new stuff.

Would this slow anything down? Nope. Same stuff generated.

Would this introduce bugs in code? Nope, this is a new keyword that is highly likely to not be used in someone's code that would cause problems.

I'm very sure that this syntax is also causing developers to write bad code (because I've seen it) doing the following:

from c in Context.Courses
join ci in Context.CourseItems on c.Id equals ci.CourseId

let ti = (from i in Context.TopicItems where i.TopicId == ci.ItemId select I)

Which results in a sub select disaster which is INCREDIBLY SLOW.

Please save us from our insanity and work with the C# team to get this keyword added and make Entity Framework sane for left outer joins! (all right outer joins can be written as lefts so no need to confuse the matter)

**Comment by SimplerSoftware** [comment]
This would be extremely helpful!

**Comment by jeffrey-igaw** [comment]
Yes! we every write left join query as below, this is not clear and not expectable what is mean.

```csharp
_context.Users
    .GroupJoin(_context.Posts, u => u.Id,  p => p.UserId, posts => posts.DefaultIfEmpty())
    .SelectMany(g => new {User = g.Key, Posts = g.Value})
    .Something();
```

maybe we want to use as ..

```csharp
_context.Users
    .LeftJoin(_context.Posts. u => u.Id, p => p.UserId, (user, post/* nullable? */) => new {user, post})
    .Where(/* something */); 
```

**Comment by roji** [comment]
FYI API proposal to introduce LeftJoin into LINQ: https://github.com/dotnet/runtime/issues/110292

**Comment by roji** [comment]
With a first-class LeftJoin() LINQ operator going into .NET 10 (and hopefully the corresponding LINQ expression syntax into C#), this issue now tracks having EF recognize the new operator. The previous construct (GroupJoin/SelectMany/DefaultIfEmpty) should still be recognized as before, but probably normalized to the new .NET LeftJoin operator internally.

Note that RightJoin is also being introduced. We can provide a default implementation that transforms it to LeftJoin, until actual support for translate RightJoin to SQL RIGHT JOIN is implemented (the latter is tracked by #35367).

**Comment by stanislav-gordenko** [comment]
Hallo!
What is the different between LeftJoin and Include?

**Comment by JamesFieldist** [comment]
LeftJoin is optional nullable and allows you to select subsets.

Include is a loading operation.
